src/Serilog/Capturing/PropertyValueConverter.cs

```diff
@@ -38,6 +40,8 @@ partial class PropertyValueConverter : ILogEventPropertyFactory, ILogEventProper
     readonly int _maximumStringLength;
     readonly int _maximumCollectionCount;
     readonly bool _propagateExceptions;
+    readonly IDictionary<Type, DestructuringFallback> _fallbackDestructuring;
+    readonly ConcurrentDictionary<Type, Destructuring?> _fallbackDestructuringCache;
```

Comment:
I'm not sure if a concurrent dictionary is necessary or if a regular one would be sufficient. I assumed that the converter is used multi-threaded.